src/setfit/utils.py

```diff
@@ -131,3 +131,24 @@ def summary(self) -> None:
         Prints summary of all benchmarks performed.
         """
         self.print(f"\n{'#' * 30}\nBenchmark Summary:\n{'#' * 30}\n\n{self.summary_msg}")
+
+
+class BestRun(NamedTuple):
+    """
+    The best run found by an hyperparameter search (see [`~Trainer.hyperparameter_search`]).
```

Comment:
should this refer to the `SetFitTrainer` or the `Trainer` from `transformers`? If the former, I'd use this:
```suggestion
    The best run found by an hyperparameter search (see [`~SetFitTrainer.hyperparameter_search`]).
```